homework-2/script/maim.js

```diff
@@ -0,0 +1,318 @@
+            `;
+            
+        }); 
+        
+        let changeDataBtn = document.querySelectorAll('.fa-pen');
+            deleteBtn = document.querySelectorAll('.delete-user');
+            secondaryCheckbox = document.querySelectorAll('.box-checkbox');
+            checkboxBtn = document.querySelectorAll('.tabel-checkbox');
+    
+        secondaryCheckbox.forEach(( item, index ) => {
+                item.addEventListener('click', () => {
+                    if ( !item.classList.contains('checked') ) {
+                        item.classList.add('checked')
+                        deleteBtn[index-1].classList.add('active-delete-btn');
+                        deleteData( deleteBtn, dataIdUser );
+                        return;
+                    } else {
+                        item.classList.remove('checked')
+                        deleteBtn[index-1].classList.remove('active-delete-btn');
+                    }
+                    
+                });
+            });
+            changeData(changeDataBtn);
+            
+    }
+    
+    checkboxFirst.addEventListener('click', (e) => {
+        if (e.target.checked) {
+            allCheckbox.classList.add('checked');
+            e.target.classList.add('checked');
+            deleteBtn.forEach(deleteBtn => deleteBtn.classList.add('active-delete-btn'));
+            checkboxBtn.forEach(checkbox => checkbox.checked = 'true');
+            secondaryCheckbox.forEach(checkBnt => checkBnt.classList.add('checked'));
+        } else {
+            allCheckbox.classList.remove('checked');
+            checkboxBtn.forEach(checkbox => checkbox.checked = 'false');
+            deleteBtn.forEach(deleteBtn => deleteBtn.classList.remove('active-delete-btn'));
+            secondaryCheckbox.forEach(checkBnt => checkBnt.classList.remove('checked'));
+        }
+    });
+
+    
+    function changeData ( btnPen ) {
+        btnPen.forEach(( button, index ) => {
+            button.addEventListener('click', (e) => {
+                modalNumber = index;
+                modal.classList.add( 'show' );
+                valueCompany.value = dataCompany[index];
+                valueName.value = dataName[index];
+                valueAddress.value = dataAddress[index];  
+                
+            });
+        });
+
+    };
+    
+
+    closeBtn.forEach((close) => {
+        close.addEventListener('click', (e) => {
+            modal.classList.remove( 'show' );
+            modalCreateUser.classList.remove('show');    
+        });
+    });
+
+    showPostForm.addEventListener('click', () => {
+        modalCreateUser.classList.add('show');
+    });
+    
+    
+    function deleteData ( btn, dataIdUser ) {
+        btn.forEach(( item, index ) => {
+            item.addEventListener('click', (e) => {
+                e.preventDefault();
+              deleteUser(dataIdUser[index]); 
+            });
+        });
+    };
+
+    updateUserBtn.addEventListener('click' , (e) => {
```

Comment:
this doesn't work, re-check update function

Reply:
they are actually working, but data is coming after reloading, needs to be fixed and you need to close the modal after save/update btn is clicked